code/game/machinery/bounty_board.dm

```diff
@@ -0,0 +1,217 @@
+				new /obj/item/wallframe/bounty_board(loc)
+			qdel(src)
+
+/obj/machinery/bounty_board/ui_interact(mob/user, ui_key, datum/tgui/ui, force_open, datum/tgui/master_ui, datum/ui_state/state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "RequestKiosk", name, ui_x, ui_y, master_ui, state)
+		ui.open()
+
+/obj/machinery/bounty_board/ui_data(mob/user)
+	var/list/data = list()
+	var/list/formatted_requests = list()
+	var/list/formatted_applicants = list()
+	for(var/i in GLOB.request_list)
+		if(!i)
+			continue
+		var/datum/station_request/request = i
+		formatted_requests += list(list("owner" = request.owner, "value" = request.value, "description" = request.description, "acc_number" = request.req_number))
+		if(request.applicants)
+			for(var/datum/bank_account/j in request.applicants)
+				formatted_applicants += list(list("name" = j.account_holder, "request_id" = request.owner_account.account_id, "requestee_id" = j.account_id))
+	var/obj/item/card/id/id_card = user.get_idcard()
+	if(id_card?.registered_account)
+		current_user = id_card.registered_account
+	if(current_user)
+		data["AccountName"] = current_user.account_holder
+	data["Requests"] = formatted_requests
+	data["Applicants"] = formatted_applicants
+	return data
+
+/obj/machinery/bounty_board/ui_act(action, list/params)
+	if(..())
+		return
+	var/current_ref_num = params["request"]
+	var/current_app_num = params["applicant"]
+	var/datum/bank_account/request_target
+	for(var/datum/station_request/i in GLOB.request_list)
+		if("[i.req_number]" == "[current_ref_num]")
+			active_request = i
+			break
+	if(active_request)
+		for(var/datum/bank_account/j in active_request.applicants)
+			if("[j.account_id]" == "[current_app_num]")
+				request_target = j
+				break
+	switch(action)
+		if("CreateBounty")
+			say("Dispensing Card.")
+			new /obj/item/bounty_card(loc)
+			return TRUE
+		if("Apply")
+			if(!current_user)
+				say("Please swipe a valid ID first.")
+				return TRUE
+			if(current_user.account_holder == active_request.owner)
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 20, TRUE)
+				return TRUE
+			active_request.applicants += list(current_user)
+		if("PayApplicant")
+			if(!current_user)
+				return
+			if(!current_user.has_money(active_request.value) || (current_user.account_holder != active_request.owner))
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+				return
+			request_target.transfer_money(current_user, active_request.value)
+			say("Paid out [active_request.value] credits.")
+			return TRUE
+		if("Clear")
+			if(current_user)
+				current_user = null
+				say("Account Reset.")
+				return TRUE
+		if("DeleteRequest")
+			if(active_request.owner != current_user.account_holder)
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 20, TRUE)
+				return TRUE
+			say("Deleted current request.")
+			GLOB.request_list.Remove(active_request)
+			return TRUE
+	. = TRUE
+
+/obj/item/bounty_card
```

Comment:
Is there any reason for these to even exist ? You could just fill all the data in ui ?

Reply:
29 days ago when this was PR'd we didn't have a longform text component in TGUi for writing inside the UI, so this was a simpler solution. I can rewrite the UI with that in mind but this will be a headache I'm calling it now.

Reply:
Mmm, So upon further inspection I was mislead, we don't actually have a longform text component, it's still the same small text input, and with the extra amount of sanity and work required for paper to get what it has working, using the bounty slips is still an easier way to circumvent that while getting input sanity

Reply:
https://github.com/tgstation/tgstation/blob/master/tgui/packages/tgui/components/TextArea.js